File: LearningRuby/facts/tree.rb
class Node
    attr_accessor  :left,:right,:data
    def initialize(data,left,right)
        @data=data
        @left=left
        @right=right
    end
    def to_s()
        s=@left.nil?() ? "." : @left.data.to_s
        s+="<-"+@data.to_s+"->"
        s+=@right.nil?() ? "." : @right.data.to_s
        return s
    end
end
def display(node)
    if(node==nil)
        return
    end
    puts(node)
    display(node.left)
    display(node.right)
end
right=Node.new(75,nil,nil)
left=Node.new(50,nil,nil)
root=Node.new(25,left,right)
display(root)

File: Learning Dart/tree.dart
import 'dispalyArr.dart';

class Node {
  int data;
  Node left;
  Node right;
  Node(int data, Node left, Node right) {
    this.data = data;
    this.left = left;
    this.right = right;
  }
}

void main(List<String> args) {
  Node right = new Node(10, null, null);
  Node left = new Node(40, null, null);
  Node root = new Node(20, left, right);
  display(root);
}

display(root) {
  if (root == null) {
    return;
  }
  var str = "";
  str += (root.left == null) ? "." : root.left.data.toString();
  str += " <- " + root.data.toString() + " -> ";
  str += (root.right == null) ? "." : root.right.data.toString();
  print(str);
  display(root.left);
  display(root.right);
}
